roadmap/implementers-guide/src/types/overseer-protocol.md

```diff
@@ -40,9 +40,9 @@ struct ActiveLeavesUpdate {
 Messages received by the approval voting subsystem.
 
 ```rust
-enum VoteCheckResult {
+enum VoteCheckResult<T> {
 	// The vote was accepted and should be propagated onwards.
-	Accepted,
+	Accepted(T),
```

Comment:
There's some documentation in `ApprovalVotingMessage` about what `T` will likely contain, but that's not where people are likely to look first when trying to figure out this enum. Copying that note here would help; expanding it would help more.

Reply:
What kind of documentation do you expect? It's a generic

Reply:
Yes, it's generic, with no locally observable constraints. That lack of information is hard to work with. I'd go with documentation like:

> In typical use, the payload of `Accepted` is either `()` when routed via `CheckAndImportApproval`, or `(Hash, u32)` when routed via `CheckAndImportAssignment`. In the latter case, those values are the block hash and candidate index of the assignment.

The goal is just to provide a local jumping-off point, so the reader can have some idea of what's going on there without needing to correlate it with the entire rest of the codebase.

Reply:
I find that to be too fragile for the moment. It would be easy to update one of those types in a small iteration without updating that doc-string, leading the reader even more confused. Whereas the `VoteImportResult` itself is easily searchable.